order/src/main/java/org/order/OrderService.java

```diff
@@ -0,0 +1,34 @@
+package org.order;
+
+import jakarta.transaction.Transactional;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import org.order.dto.PurchaseOrderDto;
+import org.order.entity.PurchaseOrder;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class OrderService {
+
+    private final OrderMapper orderMapper;
+    private final PurchaseOrderRepository purchaseOrderRepository;
+
+    @Transactional
+    Long addOrder(PurchaseOrderDto purchaseOrderDto) {
+        PurchaseOrder purchaseOrder = purchaseOrderRepository.save(orderMapper.mapOrderDtoToEntity(purchaseOrderDto));
+        return purchaseOrder.getPurchaseOrderId();
+    }
+
+    @Transactional
+    List<PurchaseOrderDto> findAll() {
+        List<PurchaseOrder> purchaseOrders = purchaseOrderRepository.findAll();
+        return purchaseOrders.stream().map(orderMapper::mapOrderEntityToDto).collect(Collectors.toList());
```

Comment:
1. it can be inline.
2. code as suggestion
```suggestion
        return purchaseOrders.stream().map(orderMapper::mapOrderEntityToDto).list();
```

Reply:
ad 1 - that's totally true, but thanks to one extra line is more explicite what type we are getting from findAll(), and to what type we are mapping it. So sure we can put it in one line, but for me it's more readable with two.